enterprise/internal/campaigns/resolvers/resolver.go

```diff
@@ -631,3 +639,19 @@ func checkSiteAdminOrSameUser(ctx context.Context, userID int32) (bool, error) {
 	}
 	return true, nil
 }
+
+type ErrInvalidFirstParameter struct {
+	Min int
+	Max int
+}
+
+func (e ErrInvalidFirstParameter) Error() string {
+	return fmt.Sprintf("invalid first param given: min=%d, max=%d", e.Min, e.Max)
+}
+
+func validateFirstParam(first int32, min, max int) error {
+	if first < int32(min) || first > int32(max) {
+		return ErrInvalidFirstParameter{Min: min}
+	}
+	return nil
+}
```

Comment:
I don't think there's a need for `min` if it's always `0`:

```suggestion
func validateFirstParam(first int32, max int) error {
	if first < 0 || first > int32(max) {
		return ErrInvalidFirstParameter{Min: 0, Max: max, First: first}
	}
	return nil
}
```

Reply:
I'd also introduce another function:

```go
const defaultMaxFirstParam = 1000

func validateFirstParamDefaults(first int32) error {
  return validateFirstParam(first, defaultMaxFirstParam)
}
```

and then use that everywhere instead of passing `0, 1000` around.